frame/contracts/src/exec.rs

```diff
@@ -268,16 +290,19 @@ where
 			Err(Error::<T>::MaxCallDepthReached)?
 		}
 
+		let contract = <ContractInfoOf<T>>::get(&dest)
+			.and_then(|contract| contract.get_alive())
+			.ok_or(Error::<T>::NotCallable)?;
+
+		let executable = E::from_storage(contract.code_hash, &self.config.schedule)?;
```

Comment:
sidenote: when the contract loader needs to do re-instrumentation of the contract, is this cost accounted in gas amount ?

The issue I see is that when schedule is changed, someone can try to call a lot of contracts just to trigger re-instrumentation, and then the block weight would be inaccurate.

(no need to fix it in this PR)

Reply:
This is not accounted for. I have no good solution for this right now. Having the first caller pay for the re-instrumentation pay in gas would be bad as this could result in a game of chicken. At least it would be surprising.  Not sure how critical this is.

Reply:
Will do this in another PR.

Reply:
to avoid having the first caller pay for the re-instrumentation in gas, maybe chains should re-instrument all contract with something like on-idle.
or also provide a mecanism for callers to share the payment of the re-instrumentation.